### ISSUE TITLE
Problems packing `.mod` files

### ISSUE THREAD

**Issue by th3w1zard1** [NickHugi/PyKotor#11]
Not sure what the cause is, but installing both the K1 and K2 modbuild with PyKotor results in all characters having no dialogue (you try to talk to them and they say nothing). Doesn't affect playable characters in the party.
Loading into a new zone (walking through a door into a new area) causes the game to return to the main menu with all buttons there disabled.  This is an instantaneous action. Attempting to alt+f4 does not even bring up the confirmation 'are you sure' screen.

I'm attempting to compare the .mod files as I think it is a lip problem, I will post my results here.

after installing TSLRCM Tweak Pack
![image](https://github.com/NickHugi/PyKotor/assets/2219836/a7b9a7ea-88a4-492e-842f-0f8898a9671f)

**Comment by th3w1zard1** [comment]
after installing TSLRCM Tweak Pack
![image](https://github.com/NickHugi/PyKotor/assets/2219836/a7b9a7ea-88a4-492e-842f-0f8898a9671f)

**Comment by NickHugi** [comment]
Yeah I can confirm from past experiences that the extension not matching whats in the fileheader does break the file when its loaded by the game.

**Comment by th3w1zard1** [comment]
I notice in TSLPatcher source the following exists for packing .mod files:
```
  $self->{'entry_count'}, # 0x20
  $self->{'offset_to_localized_string'}
```
However in PyKotor we have:
```
        self._writer.write_uint32(len(self.erf)) # 0x20
        self._writer.write_uint32(0)
```
Small inconsistency at 0x24. Not sure if it's required by the game, it's used like this in TSLPatcher source:
`$self->{'offset_to_key_list'}=$self->{'offset_to_localized_string'}+$self->{'localized_string_size'};`

from the aurora engine pdf:
![image](https://github.com/NickHugi/PyKotor/assets/2219836/cc1b59b3-aa54-4dbf-b838-0dfaa1828547)
